defs/gmake.mk

```
@@ -477,3 +477,11 @@ spec/%/ spec/%_spec.rb: programs exts PHONY
 	+$(RUNRUBY) -r./$(arch)-fake $(srcdir)/spec/mspec/bin/mspec-run -B $(srcdir)/spec/default.mspec $(SPECOPTS) $(patsubst %,$(srcdir)/%,$@)
 
 ruby.pc: $(filter-out ruby.pc,$(ruby_pc))
+
+$(COROUTINE_OBJ).json:
+	$(Q) echo > $@
+
+compile_commands.json: $(ALLOBJS) $(COROUTINE_OBJ).json
+	$(Q) echo '[' > $@
+	$(Q) cat $(ALLOBJS:.o=.o.json) >> $@
+	$(Q) echo ']' >> $@
```

And a trailing comma in an array is not allowed.

An idea is to use a sed script, say tool/json_list.sed:
```sed
#!/usr/bin/sed -f
1i\
[
$s/,$//
$a\
]
```

Then the command would be:
```suggestion
	$(Q) sed -f $(tooldir)/json_list.sed $(ALLOBJS:.o=.o.json) > $@
```

---

Ya, I know a trailing comma isn't allowed, but `clangd` seems to accept it.  Should we still add the `sed` script even if clang doesn't care?

---

I added the sed file 😄